### ISSUE TITLE
check resulting in segmentation fault

### ISSUE THREAD

**Issue by jerome-peng** [pgbackrest/pgbackrest#999]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version:  pgBackRest 2.25

2. PostgreSQL version: 12.2

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: CentOS Linux release 7.6.1810 (Core) 

4. Did you install pgBackRest from source or from a package? source

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
[demo]
pg1-path=/opt/data6000
pg1-port=6000
pg1-user=postgres

[global]
repo1-path=/home/postgres/pgbackrest

    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
port=6000
archive_command='/usr/bin/pgbackrest --stanza=demo archive-push /opt/data6000/%p'
archive_mode = on
listen_addresses = '*'
wal_level = logical
max_wal_senders=8

    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

$  pgbackrest --stanza=demo \
> --repo-path=/home/postgres/pgbackrest \
> --config=/home/postgres/pgbackrest/etc/pgbackrest.conf \
> --log-path=/home/postgres/pgbackrest/log \
> --log-level-file=info --log-subprocess \
> check 
ERROR: [082]: WAL segment 0000000200000004000000F7 was not archived before the 60000ms timeout
       HINT: check the archive_command to ensure that all options are correct (especially --stanza).
       HINT: check the PostgreSQL server log for errors.

$ tailf -n5 /opt/data6000/pg_log/pg_log_2020-04-14_153907.csv 
2020-04-14 15:51:08.510 CST,,,6550,,5e95689b.1996,16,,2020-04-14 15:39:07 CST,,0,LOG,00000,"archiver process (PID 6609) exited with exit code 1",,,,,,,,,""
2020-04-14 15:52:08.534 CST,,,6613,,5e956ba8.19d5,1,,2020-04-14 15:52:08 CST,,0,FATAL,XX000,"archive command was terminated by signal 11: Segmentation fault","The failed archive command was: /usr/bin/pgbackrest --stanza=demo archive-push /opt/data6000/pg_wal/0000000200000004000000EF.00000028.backup",,,,,,,,""
2020-04-14 15:52:08.534 CST,,,6550,,5e95689b.1996,17,,2020-04-14 15:39:07 CST,,0,LOG,00000,"archiver process (PID 6613) exited with exit code 1",,,,,,,,,""
2020-04-14 15:53:08.555 CST,,,6617,,5e956be4.19d9,1,,2020-04-14 15:53:08 CST,,0,FATAL,XX000,"archive command was terminated by signal 11: Segmentation fault","The failed archive command was: /usr/bin/pgbackrest --stanza=demo archive-push /opt/data6000/pg_wal/0000000200000004000000EF.00000028.backup",,,,,,,,""
2020-04-14 15:53:08.557 CST,,,6550,,5e95689b.1996,18,,2020-04-14 15:39:07 CST,,0,LOG,00000,"archiver process (PID 6617) exited with exit code 1",,,,,,,,,""

7. Describe the issue:
$ pgbackrest --stanza=demo \
--repo-path=/home/postgres/pgbackrest \
--config=/home/postgres/pgbackrest/etc/pgbackrest.conf \
--log-path=/home/postgres/pgbackrest/log \
--log-level-file=info --log-subprocess \
stanza-create 

$ cat pgbackrest/log/demo-stanza-create.log
...
INFO: stanza-create command end: completed successfully (534ms)

$  pgbackrest --stanza=demo \
> --repo-path=/home/postgres/pgbackrest \
> --config=/home/postgres/pgbackrest/etc/pgbackrest.conf \
> --log-path=/home/postgres/pgbackrest/log \
> --log-level-file=info --log-subprocess \
> check 
ERROR: [082]: WAL segment 0000000200000004000000F7 was not archived before the 60000ms timeout
       HINT: check the archive_command to ensure that all options are correct (especially --stanza).
       HINT: check the PostgreSQL server log for errors.

$ tailf -n6  pgbackrest/log/demo-check.log 
-------------------PROCESS START-------------------
2020-04-14 15:43:55.719 P00   INFO: check command begin 2.25: --config=/home/postgres/pgbackrest/etc/pgbackrest.conf --log-level-file=info --log-path=/home/postgres/pgbackrest/log --log-subprocess --pg1-path=/opt/data6000 --pg1-port=6000 --pg1-user=postgres --repo1-path=/home/postgres/pgbackrest --stanza=demo
2020-04-14 15:44:56.456 P00  ERROR: [082]: WAL segment 0000000200000004000000F7 was not archived before the 60000ms timeout
                                    HINT: check the archive_command to ensure that all options are correct (especially --stanza).
                                    HINT: check the PostgreSQL server log for errors.
2020-04-14 15:44:56.456 P00   INFO: check command end: aborted with exception [082]

**Comment by jerome-peng** [comment]
check trace file

[demo-check.log](https://github.com/pgbackrest/pgbackrest/files/4473970/demo-check.log)

**Comment by jerome-peng** [comment]
check trace file

[demo-check.log](https://github.com/pgbackrest/pgbackrest/files/4473976/demo-check.log)

**Comment by jerome-peng** [comment]
[082]: WAL segment 0000000200000004000000F8 was not archived before the 60000ms timeout
                                    HINT: check the archive_command to ensure that all options are correct (especially --stanza).
                                    HINT: check the PostgreSQL server log for errors.
                                    STACK TRACE:
                                    command/archive/common:walSegmentFind:453:(storage: {type: posix, path: {"/home/postgres/pgbackrest"}, write: false}, archiveId: {"12-1"}, walSegment: {"0000000200000004000000F8"}, timeout: 60000)
                                    command/check/check:checkPrimary:(dbGroup: {DbGetResult}, storage: {type: posix, path: {"/home/postgres/pgbackrest"}, write: false}, pgVersion: 120000, pgSystemId: 6789175459460411344, cipherType: 0)
                                    command/check/check:cmdCheck:(void)
                                    main:main:(debug log level required for parameters)

**Comment by dwsteele** [comment]
Well, this is a bit unusual:
```
archive_command='/usr/bin/pgbackrest --stanza=demo archive-push /opt/data6000/%p'
```
Try setting it to:
```
archive_command='/usr/bin/pgbackrest --stanza=demo archive-push %p'
```
and see what happens.

The original setting should work or give a graceful error, but if the updated archive command works that will help give us an idea of what is going on.

**Comment by jerome-peng** [comment]
First i config this

> archive_command='/usr/bin/pgbackrest --stanza=demo archive-push %p' 

i get error from pg log

> ERROR: [037]: option 'pg1-path' must be specified when relative wal paths are used
       HINT: is %f passed to archive-push instead of %p?
       HINT: PostgreSQL may pass relative paths even with %p depending on the environment.

then i change to this
> archive_command='/usr/bin/pgbackrest --stanza=demo archive-push /opt/data6000/%p'

**Comment by jerome-peng** [comment]
[pgbackrest20200415.txt](https://github.com/pgbackrest/pgbackrest/files/4478489/pgbackrest20200415.txt)

**Comment by dwsteele** [comment]
Here's the clue:
```
ERROR: [037]: option 'pg1-path' must be specified when relative wal paths are used
```
You won't be able to do a backup unless `pg1-path` is set (see https://pgbackrest.org/user-guide-centos7.html#quickstart/configure-stanza) so go ahead and do that and set your `archive_command` back to:
```
archive_command='/usr/bin/pgbackrest --stanza=demo archive-push %p'
```
We'll investigate why the other way was causing problems, but you need `pg1-path` set in any case.

**Comment by jerome-peng** [comment]
no, i set it

[demo]
pg1-path=/opt/data6000
pg1-port=6000
pg1-user=postgres

**Comment by jerome-peng** [comment]
that's why i change the "archive_command" from  "%p"  to /opt/data6000/%p""

**Comment by dwsteele** [comment]
Where is your config file? It looks like the archive command isn't seeing it.

**Comment by jerome-peng** [comment]
Please attach the following as applicable:

pgbackrest.conf 

[demo]
pg1-path=/opt/data6000
pg1-port=6000
pg1-user=postgres

[global]
repo1-path=/home/postgres/pgbackrest

**Comment by dwsteele** [comment]
> Where is your config file?

I meant, what is the path to your config file?

**Comment by jerome-peng** [comment]
i am sorry, you can see from my trace file "pgbackrest20200415.txt"

i use --config

--config=/home/postgres/pgbackrest/etc/pgbackrest.conf

**Comment by dwsteele** [comment]
OK, if you use a custom config location you'll need to pass it to *every* pgbackrest command, including `archive-push`. So:
```
archive_command='/usr/bin/pgbackrest --config=/home/postgres/pgbackrest/etc/pgbackrest.conf --stanza=demo archive-push %p'
```
It's easier to put the config file in one of the default locations (https://pgbackrest.org/user-guide-centos7.html#quickstart/configure-stanza) when possible.

**Comment by jerome-peng** [comment]
thanks, it work. 

`archive_command='/usr/bin/pgbackrest --config=/home/postgres/pgbackrest/etc/pgbackrest.conf --stanza=demo archive-push %p'`

**Comment by dwsteele** [comment]
I'm going to leave this open because the segfault is clearly a bug. My guess is there is somewhere else that wants `pg1-path` but isn't throwing an error.

**Comment by jerome-peng** [comment]
well thanks a lot.

**Comment by dwsteele** [comment]
We are working on a bug fix for the underlying problem in this issue. How would you like your name to appear in the release notes. Is "Jerome Peng" ok?

**Comment by jerome-peng** [comment]
> We are working on a bug fix for the underlying problem in this issue. How would you like your name to appear in the release notes. Is "Jerome Peng" ok?

it's ok, Jerome Peng

**Comment by dwsteele** [comment]
Fixed at 816ba9244 but forgot to add you to the release notes so did that at ef932499225.
